### PR TITLE
Add documentation to ZendeskHelper - Feedback PR

### PR DESCRIPTION
**Pull request by oguzkocer**
**Note:** This PR depends on #7911 and shouldn't be merged until #7911 is merged and the base branch is changed to `issue/zendesk-master`.

This PR adds documentation to the `ZendeskHelper`. I have been planning a small refactor for the `ZendeskHelper` and `SupportHelper`, as also mentioned in #7911, however I couldn't find any important wins in the code after a few hours spent on it trying a few different ideas. I was also planning to add some documentation to it, so I thought a separate PR for that would be good way to kick of some discussion.

@0nko @planarvoid You have been the main reviewers so far on the Zendesk PRs, I wanted to ask your opinion on whether there is anything you see that could be improved and have any suggestions on how to go about it.

Personally, the thing I wanted to improve the most is how the support identity dialog is shown through `SupportHelper`. We always depend on the `AppPrefs` for this information, so what I wanted to do was to separate the dialog from the Zendesk actions. However, Zendesk actions are called all over and the ideas I tried was only giving me small wins which didn't feel like it was good enough. I'll add one of the things I tried as a diff to showcase what I am talking about.

The other thing is, I am not sure whether adding tests for `SupportHelper` and `ZendeskHelper` would be a win for us. I have been thinking about this for some time and I can't find any value in them as we are just passing information to Zendesk singleton. Maybe there is some value in adding some tests, but I can't find a way to setup the tests that would take advantage of it. @planarvoid I know you're specifically interested in tests, so maybe you have some ideas about that? Or maybe you can at least confirm that it's not a valuable part to have tests on.

These are the two big things I had in my mind, but please feel free to bring up any concerns, suggestions etc about the Zendesk project so far. This is hopefully the last PR except for push notifications, so it's a great time to make any improvements remaining.

/cc @aerych

**P.S:** The PR is targeting a WIP branch, that's why it doesn't have a milestone.

### CODE REVIEW

**Comment by oguzkocer** [comment]
This is a diff of the initial commit for one of the ideas I was trying out (as mentioned in the PR description). It was a start to detach the dialog from the Zendesk actions.

```
diff --git a/WordPress/src/main/java/org/wordpress/android/support/SupportHelper.kt b/WordPress/src/main/java/org/wordpress/android/support/SupportHelper.kt
index 982dda7602..2a62d02fee 100644
--- a/WordPress/src/main/java/org/wordpress/android/support/SupportHelper.kt
+++ b/WordPress/src/main/java/org/wordpress/android/support/SupportHelper.kt
@@ -14,6 +14,11 @@ import org.wordpress.android.ui.prefs.AppPrefs
 import org.wordpress.android.util.validateEmail
 
 class SupportHelper {
+    val supportIdentity: Pair<String?, String?>
+        get() = Pair(AppPrefs.getSupportEmail(), AppPrefs.getSupportName())
+    val isSupportIdentityAvailable: Boolean
+        get() = !supportIdentity.first.isNullOrEmpty()
+
     /**
      * This function will check whether there is a support email saved in the AppPrefs and use the saved email and name
      * to run the provided function, most likely a support request. If there is no saved support email, it'll trigger a
@@ -29,17 +34,17 @@ class SupportHelper {
         context: Context,
         account: AccountModel?,
         selectedSite: SiteModel?,
-        emailAndNameSelected: (String, String) -> Unit
+        emailAndNameSelected: () -> Unit
     ) {
         val currentEmail = AppPrefs.getSupportEmail()
         if (!currentEmail.isNullOrEmpty()) {
-            emailAndNameSelected(currentEmail, AppPrefs.getSupportName())
+            emailAndNameSelected()
         } else {
             val (emailSuggestion, nameSuggestion) = getSupportEmailAndNameSuggestion(account, selectedSite)
             showSupportIdentityInputDialog(context, emailSuggestion, nameSuggestion, false) { email, name ->
                 AppPrefs.setSupportEmail(email)
                 AppPrefs.setSupportName(name)
-                emailAndNameSelected(email, name)
+                emailAndNameSelected()
             }
         }
     }
diff --git a/WordPress/src/main/java/org/wordpress/android/support/ZendeskHelper.kt b/WordPress/src/main/java/org/wordpress/android/support/ZendeskHelper.kt
index 9f6bc5362a..987e274586 100644
--- a/WordPress/src/main/java/org/wordpress/android/support/ZendeskHelper.kt
+++ b/WordPress/src/main/java/org/wordpress/android/support/ZendeskHelper.kt
@@ -11,7 +11,6 @@ import org.wordpress.android.fluxc.store.AccountStore
 import org.wordpress.android.fluxc.store.SiteStore
 import org.wordpress.android.login.BuildConfig
 import org.wordpress.android.ui.accounts.HelpActivity.Origin
-import org.wordpress.android.ui.prefs.AppPrefs
 import org.wordpress.android.util.AppLog
 import org.wordpress.android.util.DeviceUtils
 import org.wordpress.android.util.NetworkUtils
@@ -70,14 +69,8 @@ class ZendeskHelper(private val supportHelper: SupportHelper) {
         require(isZendeskEnabled) {
             zendeskNeedsToBeEnabledError
         }
-        val supportEmail = AppPrefs.getSupportEmail()
-        val supportName = AppPrefs.getSupportName()
-        val isIdentityAvailable = !supportEmail.isNullOrEmpty()
-        if (isIdentityAvailable) {
-            zendeskInstance.setIdentity(createZendeskIdentity(supportEmail, supportName))
-        } else {
-            zendeskInstance.setIdentity(createZendeskIdentity(null, null))
-        }
+        configureZendeskIdentity()
+        val isIdentityAvailable = supportHelper.isSupportIdentityAvailable
         val builder = HelpCenterActivity.builder()
                 .withArticlesForCategoryIds(ZendeskConstants.mobileCategoryId)
                 .withContactUsButtonVisible(isIdentityAvailable)
@@ -103,8 +96,8 @@ class ZendeskHelper(private val supportHelper: SupportHelper) {
         require(isZendeskEnabled) {
             zendeskNeedsToBeEnabledError
         }
-        supportHelper.getSupportIdentity(context, accountStore?.account, selectedSite) { email, name ->
-            zendeskInstance.setIdentity(createZendeskIdentity(email, name))
+        supportHelper.getSupportIdentity(context, accountStore?.account, selectedSite) {
+            configureZendeskIdentity()
             RequestActivity.builder()
                     .show(context, buildZendeskConfig(context, siteStore, origin, selectedSite, extraTags))
         }
@@ -121,12 +114,17 @@ class ZendeskHelper(private val supportHelper: SupportHelper) {
         require(isZendeskEnabled) {
             zendeskNeedsToBeEnabledError
         }
-        supportHelper.getSupportIdentity(context, accountStore.account, selectedSite) { email, name ->
-            zendeskInstance.setIdentity(createZendeskIdentity(email, name))
+        supportHelper.getSupportIdentity(context, accountStore.account, selectedSite) {
+            configureZendeskIdentity()
             RequestListActivity.builder()
                     .show(context, buildZendeskConfig(context, siteStore, origin, selectedSite, extraTags))
         }
     }
+
+    private fun configureZendeskIdentity() {
+        val (supportEmail, supportName) = supportHelper.supportIdentity
+        zendeskInstance.setIdentity(createZendeskIdentity(supportEmail, supportName))
+    }
 }
 
 // Helpers
```

**Comment by planarvoid** [comment]
@oguzkocer Overall I'm happy with both the Support and Zendesk helpers. I don't think it makes much sense to test them (since they are full of Zendesk specific static functions). They currently allow us to mock them and write tests for classes that use them. The way I look at this is that everything platform specific should be wrapped, injected and thin (so it doesn't have to be tested). It's not our goal to test Zendesk code. We just have to trust it. 

Long story short, from my side it's 👍

**Comment by oguzkocer** [comment]
@planarvoid I didn't realize @0nko was afk for a while. Do you mind merging this PR as is and when @0nko comes back if he has any suggestions, I'll implement it in a separate PR. Thanks!
